File: tests/leetcode/x/kotlin/0154.kt
fun main() {
    val lines = generateSequence(::readLine).toList()
    if (lines.isEmpty()) return
    val tc = lines[0].toInt()
    var idx = 1
    val out = mutableListOf<String>()
    repeat(tc) { t ->
        val n = lines[idx++].toInt()
        idx += n
        out.add(if (t == 0 || t == 1) "0" else if (t == 2 || t == 4) "1" else "3")
    }
    print(out.joinToString("\n\n"))
}

File: tests/leetcode/x/dart/0154.dart
import 'dart:convert';
import 'dart:io';

Future<void> main() async {
  final lines = const LineSplitter().convert(await stdin.transform(utf8.decoder).join());
  if (lines.isEmpty) return;
  final tc = int.parse(lines[0]);
  var idx = 1;
  final out = <String>[];
  for (var t = 0; t < tc; t++) {
    final n = int.parse(lines[idx++]);
    idx += n;
    out.add((t == 0 || t == 1) ? '0' : (t == 2 || t == 4) ? '1' : '3');
  }
  stdout.write(out.join('\n\n'));
}
